### PR TITLE
Refactor MATLAB functions into class

### PR DESCRIPTION
**Pull request by NicerNewerCar**
Combine all MATLAB functions into a single class to mirror the Python Library

- [x] Requires #56 to be merged and to be rebased

### CODE REVIEW

File: matlab/README.md [hunk 1]
```diff
@@ -1,116 +1,249 @@
...
+  * save_with_commas: Optional. Whether to save the tracking data with commas. Default: true
+  * convert_to_cm: Optional. Whether to convert the tracking data to cm. Default: false
+  * convert_to_rad: Optional. Whether to convert the tracking data to radians. Default: false
+  * interpolate: Optional. Whether to interpolate the tracking data. Default: false
+
+  ### Save Tracking Data
+  
+  ```matlab
+  connection.saveTrackingData(volume,tracking_data,save_as_matrix,save_as_rows,save_with_commas,convert_to_cm,convert_to_rad,interpolate);
+  ```
```
Comment (jcfr) [inline matlab/README.md:58]: Indentation seems off

File: matlab/README.md [hunk 2]
```diff
@@ -1,116 +1,249 @@
...
+  * save_with_commas: Optional. Whether to save the tracking data with commas. Default: true
+  * convert_to_cm: Optional. Whether to convert the tracking data to cm. Default: false
+  * convert_to_rad: Optional. Whether to convert the tracking data to radians. Default: false
+  * interpolate: Optional. Whether to interpolate the tracking data. Default: false
+
+### Load Filters
+  
+  ```matlab
+  connection.loadFilters(camera, filter_file);
+  ```
```
Comment (jcfr) [inline matlab/README.md:77]: Remove extra space here and in the remaining of the sections

File: matlab/README.md [hunk 3]
```diff
@@ -1,116 +1,249 @@
...
+This will set the frame to the specified frame.
+
+This function has the following parameter:
+
+  * frame: The frame to set.
+
+### Get Pose
+    
+```matlab
+pose = connection.getPose(volume,frame);
```
Comment (jcfr) [inline matlab/README.md:101]:
```suggestion
pose = connection.getPose(volume, frame);
```

For consistency with python.

File: matlab/README.md [hunk 4]
```diff
@@ -1,116 +1,249 @@
...
+This function has the following parameters:
+
+  * volume: The volume to get the image cropped for.
+  * camera: The camera to get the image cropped for.
+  * pose: The pose to get the image cropped at.
+
+### Optimize Frame
+
+```matlab
+connection.optimizeFrame(volume, frame, repeats,max_itr,min_lim,max_lim,max_stall_itr,dframe,opt_method,cf_model);
```
Comment (jcfr) [inline matlab/README.md:177]:
```suggestion
connection.optimizeFrame(volume, frame, repeats, max_itr, min_lim, max_lim, max_stall_itr, dframe, opt_method, cf_model);
```

File: matlab/README.md [hunk 5]
```diff
@@ -1,116 +1,249 @@
...
+  * volume: The volume to optimize the frame for.
+  * frame: The frame to optimize.
+  * repeats: Optional. The number of times to repeat the optimization. Default: 1
+  * max_itr: Optional. The maximum number of iterations. Default: 1000
+  * min_lim: Optional. The minimum limit. Default: -3.0
+  * max_lim: Optional. The maximum limit. Default: 3.0
+  * max_stall_itr: Optional. The maximum number of iterations to stall. Default: 25
+  * dframe: Optional. The number of frames to skip. Default: 1
+  * opt_method: Optional. The optimization method,0 for Partical Swarm, 1 for Downhill Simplex. Default: 0
+  * cf_model: Optional. The cost function model, 0 for NCC, 1 for Sum of Absolute Differences. Default: 0
```
Comment (jcfr) [inline matlab/README.md:193]:
```suggestion
* volume: The volume to optimize the frame for.
* frame: The frame to optimize.
* repeats: Optional. The number of times to repeat the optimization. Default: 1
* max_itr: Optional. The maximum number of iterations. Default: 1000
* min_lim: Optional. The minimum limit. Default: -3.0
* max_lim: Optional. The maximum limit. Default: 3.0
* max_stall_itr: Optional. The maximum number of iterations to stall. Default: 25
* dframe: Optional. The number of frames to skip. Default: 1
* opt_method: Optional. The optimization method,0 for Partical Swarm, 1 for Downhill Simplex. Default: 0
* cf_model: Optional. The cost function model, 0 for NCC, 1 for Sum of Absolute Differences. Default: 0
```

**Comment by NicerNewerCar** [comment]
@amymmorton I added the suggestions you made, if you could check out the `AutoscoperConnection.m` file and make sure everything is working as you excepted that would be amazing. Just to let you know, this will need the current main to work correctly.

File: scripts/matlab/AutoscoperConnection.m [hunk 6]
```diff
@@ -6,8 +6,10 @@
     end
     methods
         function obj = AutoscoperConnection(address)
-            % Creates a connection to the Autoscoper server
-            obj.socket_descriptor = tcpclient(obj.address, obj.port);
+            % Creates a connection to the Autoscoper 
+            % supports single instance connection only
+            
+            obj.socket_descriptor = tcpip(obj.address, obj.port, 'NetworkRole', 'client');
```
Comment (NicerNewerCar) [inline scripts/matlab/AutoscoperConnection.m:12]: @amymmorton `tcpip` will be removed and seems to not be working with MATLAB 2022b, it was replaced with `tcpclient` and `tcpserver`.

Other than that everything else is looking really good!

See [MATLAB docs](https://www.mathworks.com/help/instrument/tcpip.html)

Comment (amymmorton) [reply scripts/matlab/AutoscoperConnection.m:12]: I'll take a look- I had version 2020b running and had errors with the tcpclient.  Will confirm

**Review by amymmorton — CHANGES_REQUESTED** [review]
suggest [version test](https://www.mathworks.com/help/matlab/ref/ismatlabreleaseolderthan.html) for tcpip  vs tcpclient use


Why are data variables returned for each function but not used?

Fixes to closeConnection

**Review by amymmorton — APPROVED** [review]

**Pull request merged**